src/ec/packages/ec-component-accordion2/accordion2.story.js

```diff
@@ -3,32 +3,39 @@ import { storiesOf } from '@storybook/html';
 import { withKnobs, text } from '@storybook/addon-knobs';
 import { withNotes } from '@ecl-twig/storybook-addon-notes';
 import withCode from '@ecl-twig/storybook-addon-code';
-
+import { buttonLabels, getExtraKnobs } from '@ecl-twig/story-utils';
 import defaultSprite from '@ecl/ec-resources-icons/dist/sprites/icons.svg';
 import demoData from './demo/data';
 
 import accordion2 from './ecl-accordion2.html.twig';
 import notes from './README.md';
 
+const PrepareAccordion2 = data => {
+  data.items.forEach((item, index) => {
+    const { content, toggle } = item;
+
+    item.toggle.label = text(
+      `toggle.label ${index}`,
+      toggle.label,
+      buttonLabels.required
+    );
+
+    item.content = text(`content ${index}`, content, buttonLabels.required);
+    item.toggle.icon.path = defaultSprite;
+  });
+  getExtraKnobs(data);
+  return data;
+};
+
 storiesOf('Components/Accordion2', module)
   .addDecorator(withKnobs)
   .addDecorator(withCode)
   .addDecorator(withNotes)
   .add(
     'default',
     () => {
-      // This needs to be in the scope of this function.
-      // Called on knob's change of value.
-      demoData.items.forEach(item => {
-        const { id, content, toggle } = item;
-
-        item.toggle.label = text(`Label ${id}`, toggle.label);
-        item.content = text(`Content ${id}`, content);
-
-        item.toggle.icon.path = defaultSprite;
-      });
-
-      return accordion2(demoData);
+      const accordion2DemoData = PrepareAccordion2(demoData);
```

Comment:
Basing on the work @papegaill made, apparently in the easiest scenario, which this look like, we can avoid the separate call for the prepare function for then returning the component. You can see an example of what i'm saying in the messages component.
Not a needed change in this case, but since it looks like a simplified solution, the one that @papegaill applied, maybe we can opt for that when we don't need to further customise the data in the story we add.